plugins/source/scheduler_dfs.go

```diff
@@ -134,8 +137,29 @@ func (p *Plugin) resolveTableDfs(ctx context.Context, table *schema.Table, clien
 		}
 	}()
 
+	// we also check for PK duplicates
+	reportedPK := false
 	for r := range res {
-		p.resolveResourcesDfs(ctx, table, client, parent, r, resolvedResources, depth)
+		pk := p.resolveResourcesDfs(ctx, table, client, parent, r, resolvedResources, depth)
+		if pk != nil && !reportedPK {
+			reportedPK = true
+			pkSpec := "(" + strings.Join(table.PrimaryKeys(), ",") + ")"
+
+			// log err
+			p.logger.Error().
+				Str("table", table.Name).
+				Str("pk", pkSpec).
+				Str("value", *pk).
+				Msg("duplicate primary key")
+
+			// send to Sentry only once per table,
+			// to avoid sending too many duplicate messages
+			sentry.WithScope(func(scope *sentry.Scope) {
+				scope.SetTag("table", table.Name)
+				scope.SetTag("pk", pkSpec)
+				sentry.CurrentHub().CaptureMessage("duplicate primary key " + pkSpec + " in " + table.Name)
+			})
+		}
```

Comment:
How does the plugin version get set? Is it set during initialization?

Reply:
https://github.com/cloudquery/plugin-sdk/blob/main/serve/source.go#L142